rrtypes.go

```diff
@@ -0,0 +1,374 @@
+		Name: n.Name,
+		TTL:  n.TTL,
+		Type: "NS",
+		Data: n.Target,
+	}
+}
+
+// SRV represents a parsed SRV-type record, which is used to
+// manifest services or instances that provide services on a
+// network.
+//
+// The serialization of this record type takes the form:
+//
+//	_service._proto.name. ttl IN SRV priority weight port target.
+//
+// Note that all fields are mandatory.
+type SRV struct {
+	// “Service” is the name of the service being offered, without the leading
+	// underscore. The correct value for this field is defined by the service
+	// that you are serving (and is typically registered with IANA). Some
+	// examples include "sip", "xmpp", "ldap", "minecraft", "stun", "turn", etc.
+	Service string
+
+	// “Transport” is the name of the transport protocol used by the service,
+	// without the leading underscore. This is almost always "tcp" or "udp", but
+	// "sctp" and "dccp" are technically valid as well.
+	//
+	// Note that RFC 2782 defines this field as “Proto[col]”, but we're using
+	// the updated name “Transport” from RFC 6335 in order to avoid confusion
+	// with the similarly-named field in the SVCB record type.
+	Transport string
+
+	Name     string
+	TTL      time.Duration
+	Priority uint16 // Lower values indicate that clients should prefer this server
+	Weight   uint16 // Higher values indicate that clients should prefer this server when choosing between targets with the same priority
+	Port     uint16 // The port on which the service is running.
+	Target   string // The hostname of the server providing the service, which must not point to a CNAME.
+}
+
+func (s SRV) RR() RR {
+	var name string
+	if s.Service == "" && s.Transport == "" {
+		// If both “Service” and “Transport” are empty, then we'll assume that
+		// “Name” is complete as-is. This is fairly dubious, but could happen
+		// if a properly-underscored CNAME points at a SRV without underscores.
+		name = s.Name
+	} else {
+		// Otherwise, we need to prepend the underscores to the name.
+		name = fmt.Sprintf("_%s._%s.%s", s.Service, s.Transport, s.Name)
+	}
+
+	return RR{
+		Name: name,
+		TTL:  s.TTL,
+		Type: "SRV",
+		Data: fmt.Sprintf("%d %d %d %s", s.Priority, s.Weight, s.Port, s.Target),
+	}
+}
+
+// SVCB represents a parsed SVCB-type record, which is used to provide the
+// target and various key–value parameters for a service. HTTPS records are
+// defined as a “SVCB-Compatible RR Type”, which means that their data
+// structures are identical to SVCB records, albeit with a different type name
+// and semantics.
+//
+// HTTPS-type records are  used to provide clients with information for
+// establishing HTTPS connections to servers. It may include data about ALPN,
+// ECH, IP hints, and more.
+//
+// Unlike the other RR types that are hostname-focused or service-focused, SVCB
+// (“Service Binding”) records are URL-focused. This distinction is generally
+// irrelevant, but is important when disusing the port fields.
+type SVCB struct {
```

Comment:
I'm still having a hard time swallowing this as the catch-all type for SVCB, HTTPS, and potentially other types in the future.

Since this type is supposed to represent multiple record types, what if we did like you suggested with A/AAAA :arrow_right: "Address", and call this something more general like "ServiceBinding"? Like, I just feel weird knowing that I'll be working with HTTPS records, but would have to keep reasoning about this "SVCB" type.

Great godoc comments, though, really helpful for DNS noobs like me.

Reply:
> I'm still having a hard time swallowing this as the catch-all type for SVCB, HTTPS, and potentially other types in the future.

Yeah, I wish that they just made a single `SVCB` RR type, instead of defining a `HTTPS` RR type that's almost the same as `SVCB` but not quite. 

> Since this type is supposed to represent multiple record types, what if we did like you suggested with A/AAAA ➡️ "Address", and call this something more general like "ServiceBinding"?

Excellent idea!